### PR TITLE
[JENKINS-42840] Slave to Agent Renaming

### PR DESCRIPTION
**Pull request by kuisathaverat**
[JENKINS-42840](https://issues.jenkins-ci.org/browse/JENKINS-42840)

* Change slave.jar to remoting.jar
* Change javadoc
* Change some methods name (startSlave, copySlaveJar)
* Change messages

Pending to change SSH Slaves Plugin to something else, these are the proposals:

* SSH Nodes Plugins
* SSH Agent Nodes Plugin

### CODE REVIEW

File: src/main/java/hudson/plugins/sshslaves/SSHLauncher.java [hunk 1]
```diff
@@ -1034,15 +1036,15 @@ private void expandChannelBufferSize(Session session, TaskListener listener) {
...
      * @param listener         The listener.
-     * @param workingDirectory The directory into whihc the slave jar will be copied.
+     * @param workingDirectory The directory into whihc the agent jar will be copied.
      *
      * @throws IOException If something goes wrong.
      */
-    private void copySlaveJar(TaskListener listener, String workingDirectory) throws IOException, InterruptedException {
-        String fileName = workingDirectory + "/slave.jar";
+    private void copyAgentJar(TaskListener listener, String workingDirectory) throws IOException, InterruptedException {
+        String fileName = workingDirectory + SLASH_AGENT_JAR;
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: Note that changing file name is a breaking change for some edge use-cases. If one passes a command prefix, he/she may be able to specify path to another JAR location. I was using this trick a lot at my previous job to specify custom Remoting versions

**Comment by oleg-nenashev** [comment]
Never. SSH Agent is an existing term, which has nothing to do with Jenkins agent.

**Comment by oleg-nenashev** [comment]
I support renaming tho

**Comment by jvz** [comment]
Oh right, that could get confusing.

**Pull request merged**